src/tests/psyGen_test.py

```diff
@@ -810,3 +815,794 @@ def test_invalid_reprod_pad_size():
             "REPROD_PAD_SIZE in config.py should be a positive "
             "integer") in str(excinfo.value)
     config.REPROD_PAD_SIZE = keep
+
+
+def test_argument_dependent_arg():
+    '''Check that the dependent_arg method returns the appropriate boolean
+    value for arguments with combinations of read and write access'''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "4.5_multikernel_invokes.f90"),
+                           distributed_memory=False, api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    schedule = invoke.schedule
+    arg_f1_inc_1 = schedule.children[0].children[0].arguments.args[0]
+    arg_f1_inc_2 = schedule.children[2].children[0].arguments.args[0]
+    arg_f2_read_1 = schedule.children[0].children[0].arguments.args[2]
+    arg_f2_inc = schedule.children[1].children[0].arguments.args[0]
+    arg_f2_read_2 = schedule.children[2].children[0].arguments.args[1]
+    # different names returns False
+    assert not arg_f2_inc._dependent_arg(arg_f1_inc_1)
+    # same name both reads returns False
+    assert not arg_f2_read_1._dependent_arg(arg_f2_read_2)
```

Comment:
My trouble is that my understanding of dependencies was (and is) coloured by the Habakkuk work where a variable has a dependency on *another* variable if the latter appears on the RHS of an expression that assigns to the former.
I also know that I don't know (to quote Al Gore?) the terminology used in Computer Science when describing this type of thing. I'm assuming that you do and that's what you've used. I would find it slightly easier to understand if `_depends_arg` (which returns a boolean rather than an argument) was renamed, e.g. to `_depends_upon()`.
It would also help if you elaborated on what you mean by "dependency between" in the doc-string of that method.
Apologies if I'm being dense but I figure if that if it's not very clear to me then it won't be to (at least some) other people either. (I find this kind of thing very hard - how much knowledge do we assume and what do we spell out?)